examples/ex-batched-matrix-multiply.cpp

```diff
@@ -27,264 +27,469 @@
 /*
  *  Batched Matrix Multiply Example
  *
- *  This example carries out matrix multiplication
- *  for NMAT matrices. Here we assume
- *  matrices are small (3 x 3).
- *  Each iteration of the RAJA loop
- *  multiplies a matrix from the batch.
+ *  Example carries out batched matrix multiplication
+ *  for matrices of dimension 3 x 3 using two different
+ *  data layouts. 
+ * 
+ *  Matrices are stored in arrays A, and B. Results 
+ *  are stored in C. We introduce the notation A^{e}_rc
+ *  to correspond to the matrix entry in the row - r, 
+ *  column - c of matrix - e. Below we describe the potential
+ *  layouts in the case of two matrices NMAT=2.
  *
- *  The notation A^{e}_rc corresponds to the matrix entry
- *  in the row - r, column - c of matrix - e.
- *  
- *  Furthermore, we explore performance for two data layouts.
- *  Layout 1: Assumes matrices are contiguous in memory 
- *  enabling vectorized operations.
- *  i.e. [A^{0}_{00}, A^{0}_{01}, A^{0}_{02}, ...]
+ * Layout 1:
+ * Matrix entries are grouped together so that each 
+ * matrix is in a row major ordering. 
+ * i.e. A = [A^{0}_{00}, A^{0}_{01}, A^{0}_{02},
+ *           A^{0}_{10}, A^{0}_{11}, A^{0}_{12},
+ *           A^{0}_{20}, A^{0}_{21}, A^{0}_{22},
+ *           A^{1}_{00}, A^{1}_{01}, A^{1}_{02},
+ *           A^{1}_{10}, A^{1}_{11}, A^{1}_{12},
+ *           A^{1}_{20}, A^{1}_{21}, A^{1}_{22}];
  *
- * Layout 2: Multiplies matrices assuming matrix entries are grouped together
- * allowing for coalesced reads and writes.
- * i.e. [A^{0}_{00}, A^{1}_{00}, A^{2}_{00}, \dots]
+ * Layout 2:
+ * Matrix entries are first ordered by matrix number,
+ * then by column number, and finally by row number. 
+ * i.e. A = [A^{0}_{00}, A^{1}_{00}, A^{0}_{01},
+ *           A^{1}_{01}, A^{0}_{02}, A^{1}_{02},
+ *           A^{0}_{10}, A^{1}_{10}, A^{0}_{11},
+ *           A^{1}_{11}, A^{0}_{12}, A^{1}_{12},
+ *           A^{0}_{20}, A^{1}_{20}, A^{0}_{21},
+ *           A^{1}_{21}, A^{0}_{22}, A^{1}_{22}];
  *
- * We expect that layout 1 will perform better on the CPU
- * and layout 2 would perform better on the GPU.
+ * Since layout 1 has the entries for a matrix
+ * close in memory it simplifies vector operations. 
```

Comment:
More correctly, it 'allows' vector operations. Do we see SIMD actually generated by compilers?